core/arch/x86/x86.asm

```diff
@@ -567,7 +567,7 @@ GLOBAL_LABEL(cleanup_and_terminate:)
         mov      [4*ARG_SZ + REG_XBP], ARG4
 #else
         mov      REG_XBP, REG_XSP
-# if defined(MACOS) && !defined(X64)
+# ifdef UNIX
```

Comment:
So this changes the 64-bit behavior: is that intentional?  You found that the 64-bit alignment was off too?

Reply:
This affects only 32-bit, it is the `#else` of `#ifdef X64`.

Reply:
OK, you can see how from the PR diff view it looks suspect.